### PR TITLE
Feat/attribution js components

### PR DESCRIPTION
**Pull request by giannif**
JS SDK gets attribution 
Screen Shot 2020-03-19 at 11 36 06 AM

### CODE REVIEW

File: packages/components/src/components/gif.tsx [hunk 1]
```diff
@@ -27,153 +31,233 @@ export type EventProps = {
     // fired when the gif is right clicked
     onGifRightClick?: (gif: IGif, e: Event) => void
 }
+
+function useMutableRef<T>(initialValue?: T) {
```
Comment (giannif) [inline packages/components/src/components/gif.tsx]: There's a typescript error in Preact. Making my own `useRef`

**Comment by giannif** [comment]
Moat is also in here, since I mostly cut and pasted the `Gif` component from the `react-components` lib. Last time this was updated Preact didn't have hooks, but now we do

File: packages/components/src/components/attribution/avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    object-fit: cover;
+    width: 36px;
+    height: 36px;
+    margin-right: 8px;
+`
+type Props = { user: IUser; className?: string }
+const Avatar = ({ user, className = '' }: Props) => {
+    const url = user.avatar_url
+        ? getSmallAvatar(user.avatar_url)
+        : `https://media.giphy.com/avatars/default${Math.floor(Math.random() * 5) + 1}.gif`
```
Comment (thighhighcrocs) [inline packages/components/src/components/attribution/avatar.tsx]: Will the avatar change anytime this re-renders? If so, maybe we want to have this in state?

Comment (giannif) [reply packages/components/src/components/attribution/avatar.tsx]: @thighhighcrocs oh good catch!

**Pull request merged**